### PR TITLE
[Plugins/system/debug] Some improvements and cleanups in plugins/system/debug

### PR DESCRIPTION
**Pull request by frankmayer**
### Summary of Changes
Some improvements and cleanups in plugins/system/debug
- Make use of $this->db instead of JFactory::getDbo();
- Added missing docblocks
- Type-safety
- replace unnecessary strlen() call with simple check
- CS

### Testing Instructions
Code review
Quick test of debugger

### CODE REVIEW

**Comment by zero-24** [comment]
> Make use of $this->db instead of JFactory::getDbo();

To be sure please also implement the fallback like for $this->app :)

**Comment by andrepereiradasilva** [comment]
pleaase chekc drone code style issues
http://213.160.72.75/joomla/joomla-cms/5868

File: plugins/system/debug/debug.php [hunk 1]
```diff
@@ -258,7 +272,7 @@ public function onAfterRespond()
 		$html[] = "<script>function toggleContainer(name)
 		{
 			var e = document.getElementById(name);// MooTools might not be available ;)
-			e.style.display = (e.style.display == 'none') ? 'block' : 'none';
+			e.style.display = (e.style.display === 'none') ? 'block' : 'none';
```
Comment (Quy) [inline plugins/system/debug/debug.php]: Remove parentheses

File: plugins/system/debug/debug.php [hunk 2]
```diff
@@ -370,7 +384,7 @@ public static function removeDisplayCallback($name)
 	/**
 	 * Method to check if the current user is allowed to see the debug information or not.
 	 *
-	 * @return  boolean  True is access is allowed.
+	 * @return  boolean  True  is access is allowed.
```
Comment (Quy) [inline plugins/system/debug/debug.php]: Should be `True if`

File: plugins/system/debug/debug.php [hunk 3]
```diff
@@ -880,31 +896,31 @@ protected function displayQueries()
...
 				else
 				{
-					$barClass = 'bar-success';
+					$barClass   = 'bar-success';
 					$labelClass = 'label-success';
 				}
 
 				// Computes bargraph as follows: Position begin and end of the bar relatively to whole execution time.
 				// TODO: $prevBar is not used anywhere. Remove?
 				$prevBar = ($id && isset($bars[$id - 1])) ? $bars[$id - 1] : 0;
```
Comment (Quy) [inline plugins/system/debug/debug.php]: Remove parathensis

**Comment by frankmayer** [comment]
I don't get why codesniffer still fails...

File: plugins/system/debug/debug.php (hunk 3 above)
Comment (frankmayer) [inline plugins/system/debug/debug.php]: Thanks

File: plugins/system/debug/debug.php [hunk 4]
```diff
@@ -1564,7 +1584,9 @@ protected function displayLanguageFilesLoaded()
 
 		$html[] = '<ul>';
 
-		foreach (JFactory::getLanguage()->getPaths() as /* $extension => */ $files)
+		foreach (JFactory::getLanguage()->getPaths() as /* $extension => */
+		         $files
```
Comment (Quy) [inline plugins/system/debug/debug.php]: I would keep it on one line.

Comment (frankmayer) [inline plugins/system/debug/debug.php]: my formatter seems a bit off.. thanks

File: plugins/system/debug/debug.php [hunk 5]
```diff
@@ -384,7 +384,7 @@ public static function removeDisplayCallback($name)
 	/**
 	 * Method to check if the current user is allowed to see the debug information or not.
 	 *
-	 * @return  boolean  True  is access is allowed.
+	 * @return  boolean  True  if access is allowed.
```
Comment (Quy) [inline plugins/system/debug/debug.php]: Remove space

File: plugins/system/debug/debug.php [hunk 6]
```diff
@@ -635,63 +649,63 @@ protected function displayProfileInformation()
...
 			else
 			{
-				$barClassMem = 'bar-warning';
+				$barClassMem   = 'bar-warning';
 				$labelClassMem = 'label-warning';
 			}
 
-			$barClass .= " progress-$barClass";
+			$barClass    .= " progress-$barClass";
 			$barClassMem .= " progress-$barClassMem";
```
Comment (Quy) [inline plugins/system/debug/debug.php]: Single quote.

Comment (frankmayer) [inline plugins/system/debug/debug.php]: You mean single quote and do concatenation of the variables?

Comment (Quy) [inline plugins/system/debug/debug.php]: Nevermind.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on fc7225489ea98faafbf20207fe18e2cef5e02b92<br /><br />Code revoew<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16674">issues.joomla.org/tracker/joomla-cms/16674</a>.</sub>

File: plugins/system/debug/debug.php [hunk 7]
```diff
@@ -370,7 +384,7 @@ public static function removeDisplayCallback($name)
 	/**
 	 * Method to check if the current user is allowed to see the debug information or not.
 	 *
-	 * @return  boolean  True is access is allowed.
+	 * @return  boolean  True  if access is allowed.
```
Comment (andrepereiradasilva) [inline plugins/system/debug/debug.php]: remove extra space

File: plugins/system/debug/debug.php [hunk 8]
```diff
@@ -1110,8 +1130,8 @@ protected function displayQueries()
 
 				$htmlAccordions = JHtml::_(
 					'bootstrap.startAccordion', 'dbg_query_' . $id, array(
-						'active' => $info[$id]->hasWarnings ? ('dbg_query_explain_' . $id) : ''
-					)
+						                          'active' => $info[$id]->hasWarnings ? ('dbg_query_explain_' . $id) : '',
+					                          )
```
Comment (andrepereiradasilva) [inline plugins/system/debug/debug.php]: remove extar spaces

**Review by andrepereiradasilva — COMMENTED** [review]
small things

File: plugins/system/debug/debug.php (hunk 8 above)
Comment (frankmayer) [inline plugins/system/debug/debug.php]: Thanks :)

**Comment by andrepereiradasilva** [comment]
I have tested this item :white_check_mark: successfully on a9a68769abca686be73210a2c556b98c4014b01d<br /><br />Code review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16674">issues.joomla.org/tracker/joomla-cms/16674</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on a9a68769abca686be73210a2c556b98c4014b01d<br /><br />Code review.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16674">issues.joomla.org/tracker/joomla-cms/16674</a>.</sub>

**Comment by frankmayer** [comment]
@franz-wohlkoenig can we RTC this? Thanks

**Comment by ghost** [comment]
RTC after two successful tests.

Thanks for Hint, @frankmayer

**Pull request merged**